packages/components/src/custom-select-control/index.js

```diff
@@ -125,58 +173,51 @@ export default function CustomSelectControl( {
 					{ label }
 				</label>
 			) }
-			<Button
-				{ ...getToggleButtonProps( {
-					// This is needed because some speech recognition software don't support `aria-labelledby`.
-					'aria-label': label,
-					'aria-labelledby': undefined,
-					className: 'components-custom-select-control__button',
-					isSmall: true,
-					describedBy: getDescribedBy(),
-				} ) }
-			>
+			<Button { ...toggleButtonProps }>
 				{ itemToString( selectedItem ) }
 				<Icon
 					icon={ chevronDown }
 					className="components-custom-select-control__button-icon"
 				/>
 			</Button>
-			<ul { ...menuProps }>
-				{ isOpen &&
-					items.map( ( item, index ) => (
-						// eslint-disable-next-line react/jsx-key
-						<li
-							{ ...getItemProps( {
-								item,
-								index,
-								key: item.key,
-								className: classnames(
-									item.className,
-									'components-custom-select-control__item',
-									{
-										'is-highlighted':
-											index === highlightedIndex,
-										'has-hint': !! item.__experimentalHint,
-									}
-								),
-								style: item.style,
-							} ) }
-						>
-							{ item.name }
-							{ item.__experimentalHint && (
-								<span className="components-custom-select-control__item-hint">
-									{ item.__experimentalHint }
-								</span>
-							) }
-							{ item === selectedItem && (
-								<Icon
-									icon={ check }
-									className="components-custom-select-control__item-icon"
-								/>
-							) }
-						</li>
-					) ) }
-			</ul>
+			<OptionList isOpen={ isOpen } anchorRef={ anchorRef }>
+				<ul { ...menuProps }>
+					{ isOpen &&
```

Comment:
I wonder if the `ul` should instead be wrapped around `OptionList`. That would also avoid the `isOpen &&` check 

i.e

```jsx
<ul { ...menuProps }>
  <OptionList isOpen={ isOpen } anchorRef={ anchorRef }>
    { items.map( ... ) }
  </OptionList>
</ul>
```

Reply:
The `isOpen` check is slightly odd, but unfortunately I don't think we can move `OptionList` to be a child of the `ul`, as then the `Popover` divs would be direct descendants of the `ul` element (where only `li` / script tags [are allowed](https://html.spec.whatwg.org/multipage/grouping-content.html#the-ul-element)).